### PR TITLE
Docs writing guide

### PR DESCRIPTION
**Pull request by sodic**

### CODE REVIEW

**Review by infomiho — APPROVED** [review]

File: web/docs/writingguide.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- **Describe the _problem_ first, then the solution.** Before showing how a feature works, it's important to explain why it exists. Otherwise, users won't have the context to know if this information is important to them (is it a problem they experience?) or what prior knowledge/experience to connect it to.
+- **While writing, don't be afraid to ask questions**, _especially_ if you're afraid they might be "dumb". Being vulnerable is hard, but it's the only way for us to more fully understand what we need to explain.
+
+## Organization
+
+### Organization of pages
+
+- **Getting Started**:
+  - **Introduction** - Provides a less than 10-minute overview of the problems the project solves and why it exists.
```
Comment (Martinsos) [inline web/docs/writingguide.md:31]: s/project/Wasp ?

File: web/docs/writingguide.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+Each feature page is divided into two sections: **The guide**, and **The API reference**.
+
+#### Guide
+
+The guide tells a story about a feature and takes the reader through a step-by-step process to get the feature working.
+It goes through the feature's most important parts and does not have to be exhaustive. The goal is to provide the 20% of knowledge that will help users handle 80% of use cases.
+
+The guide is almost a tutorial, the only difference being that it can assume some context for the rest of the application.
+The text can can link to parts of the API reference, though, in most cases, you should avoid such links.
```
Comment (Martinsos) [inline web/docs/writingguide.md:58]: https://www.youtube.com/watch?v=rVSFe--rTC4

File: web/docs/writingguide.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+- **Never assume a more advanced context than you have to.**
+- **In most cases, prefer links between sections of the docs over repeating the same content in multiple sections.** Some repetition in content is unavoidable and even essential for learning. However, too much repetition also makes the docs more difficult to maintain, because a change in the API will require changes in many places and it's easy to miss something. This is a difficult balance to strike.
+- **Specific is better than generic.** For example, a `<BlogPost>` component example is better than `<ComponentA>`.
+- **Relatable is better than obscure.** For example, a `<BlogPost>` component example is better than `<CurrencyExchangeSettings>`.
+- **Be emotionally relevant.** Explanations and examples that relate to something people have experience with and care about will always be more effective.
+- **Always prefer simpler language over complex or jargony language.** For example:
+  - "You can define an Action by declaring it in Wasp." instead of "In order to define an Action, it must first be defined via a Wasp declaration"
```
Comment (Martinsos) [inline web/docs/writingguide.md:103]: Hm although here I would argue that the second sentence carries more info, it tells that first you need to do "Wasp declaration" and then something else, while the first one sounds like you just need to declare it in Wasp, which is not correct, you also need to provide JS implementation.

File: web/docs/writingguide.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  - "function that returns a function" instead of "higher order function" (while technically correct and more concise, higher order function requires knowledge and context the reader doesn't necessarily have)
+- **Avoid language that invalidates struggle**, such as "easy", "just", "obviously", etc. For reference, see [Words To Avoid in Educational Writing](https://css-tricks.com/words-avoid-educational-writing/).
+
+### Grammar
+
+- **Don't use emojis (except in discussions).** Emojis are cute and friendly, but they can be a distraction in documentation and some emoji even convey different meanings in different cultures. They also make the documentation seem unprofessional and of lower quality.
```
Comment (Martinsos) [inline web/docs/writingguide.md:109]: :sob:

File: web/docs/writingguide.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  A time offset is beneficial because it removes the text from your short-term memory, helping you view it more objectively.
+- **It's OK to ask AI to improve your text.** Just make sure to check it and correct. You should always sign off on the last version.
+- **When someone reports a problem, there is almost always a problem**, even if the solution they proposed isn't quite right. Keep asking follow-up questions to learn more.
+- People need to feel safe asking questions when contributing/reviewing content. Here's how you can do that:
+  - **Thank people for their contributions/reviews, even if you're feeling grumpy.** For example:
+    - "Great question!"
+    - "Thanks for taking the time to explain. 🙂"
+    - "This is actually intentional, but thanks for taking the time to contribute. 😊"
+  - **Listen to what people are saying and mirror if you're not sure you're understanding correctly.** This can help validate people's feelings and experiences, while also understanding if _you're_ understanding _them_ correctly.
+  - **Use a lot of positive and empathetic emojis.** It's always better to seem a little strange than mean or impatient.
```
Comment (Martinsos) [inline web/docs/writingguide.md:150]: :tada: :smile: :tada: !!!

File: web/docs/writingguide.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  - **Kindly communicate rules/boundaries.** If someone behaves in a way that's abusive/inappropriate, respond only with kindness and maturity, but also make it clear that this behavior is not acceptable and what will happen (according to the code of conduct) if they continue behaving poorly.
+- **All docs must go through the review cycle, preferably with more than a single reviewer.** Different people focus on different things. Some of us are great at coming up with examples, others easily come up with analogies and explain complex topics, some have a clear and concise writing style, etc. Therefore, try to get at least two or three people to review your document.
+
+## Possible improvements
+
+- Some parts of our docs don't follow all the guidelines outlined in this document. There's no need to start fixing all the issues right away, we can slowly improve the docs as we edit them.
+- We've talked about having a git repo with all the example code so it's easier to copy-paste, test, and maintain.
```
Comment (Martinsos) [inline web/docs/writingguide.md:158]: Is this about example apps? Those are kind of separated from docs though, right? Or is is this about something else?

**Pull request merged**